src/Grpc.AspNetCore.Server/Internal/CallHandlers/ClientStreamingServerCallHandler.cs

```diff
@@ -31,14 +33,24 @@ internal class ClientStreamingServerCallHandler<TService, TRequest, TResponse> :
         where TService : class
     {
         private readonly ClientStreamingServerMethod<TService, TRequest, TResponse> _invoker;
+        private readonly IReadOnlyList<InterceptorRegistration> _interceptors;
```

Comment:
Move to protected property on the base class. Also property should be `List<InterceptorRegistration>` to avoid interface dispatch overhead of IReadOnlyList

Reply:
Or just use the ServiceOptions on the base class.